modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -126,14 +130,20 @@ public static HttpRequestBase buildHeaders(HttpRequestBase httpRequestBase,
 	}
 
 	public static Map<String, String> postData(EndPointInfo endPointInfo,
-	                                           Map<String, String> headers)
+											   Map<String, String> headers)
 			throws IDPTokenManagerException {
 
 		HTTP_METHODS httpMethod = endPointInfo.getHttpMethod();
 		String url = endPointInfo.getEndPoint();
-		JSONObject params = endPointInfo.getRequestParams();
+
+		String params = "";
+
+		if(endPointInfo.getRequestParams() != null){
```

Comment:
Do we validate if the endPointInfo instance is null or not anywhere in the upstream code? If not, you might need to add a null check too as part of the condition configured within the if statement.